ocaml/xapi/xapi_clustering.ml

```diff
@@ -60,14 +59,16 @@ let ip_of_pif (ref,record) =
     {- that the PIF has disallow_unplug set}
     }*)
 let assert_pif_prerequisites pif =
+  let (ref, record) = pif in
   let assert_pif_permaplugged (ref,record) =
     if not record.API.pIF_disallow_unplug then
       raise Api_errors.(Server_error (pif_allows_unplug, [ record.API.pIF_uuid ] ));
     if not record.pIF_currently_attached then
       raise Api_errors.(Server_error (required_pif_is_unplugged, [ record.API.pIF_uuid ] ))
   in
   assert_pif_permaplugged pif;
-  ignore(ip_of_pif pif)
+  ignore (ip_of_pif pif);
+  debug "Got IP %s for PIF %s" record.API.pIF_IP (Ref.string_of ref)
 
 let handle_error = function
   | InternalError message -> raise Api_errors.(Server_error (internal_error, [ message ]))
```

Comment:
Why `Unix` errors are using a `failwith` instead of also raising an internal error?

Reply:
Also below in `assert_cluster_host_can_be_created` there is a failwith instead of a proper error

Reply:
the unix errors are sort of internal errors as well, except for things like host unreachable but those need to be addressed on a case by case basis, adding a unix error to the api errors wouldn't help